### PR TITLE
Use distance unit from project's CRS for Stakeout panel

### PR DESCRIPTION
**Pull request by iiLubos**
CU-861nbagwq
https://wishlist.merginmaps.com/p/yards-as-unit-for-stakeout

In case of distance in feet:
image

### CODE REVIEW

File: app/inpututils.cpp [hunk 1]
```diff
@@ -160,6 +160,35 @@ QString InputUtils::formatNumber( const double number, int precision )
...
+QString InputUtils::formatDistanceInProjectUnit( const double distanceInMeters, int precision )
+{
+  QgsCoordinateReferenceSystem projectCrs = QgsProject::instance()->crs();
+
+  switch ( projectCrs.mapUnits() )
+  {
+    case Qgis::DistanceUnit::Meters:
+      return QString( "%1 %2" ).arg( QString::number( distanceInMeters, 'f', precision ), "m" );
+    case Qgis::DistanceUnit::Kilometers:
+      return QString( "%1 %2" ).arg( QString::number( distanceInMeters / 1000, 'f', precision ), "km" );
```
Comment (PeterPetrik) [inline app/inpututils.cpp:172]: Whatabout `QgsDistanceArea:convertLengthMeasurement( distance,  projectCrs.mapUnits() );`

(https://github.com/MerginMaps/input/blob/91519b98a9d473155b5663923b14939bd2d9acda/app/inpututils.cpp#L1617C20-L1617C20)

File: app/inpututils.cpp [hunk 2]
```diff
@@ -160,6 +160,35 @@ QString InputUtils::formatNumber( const double number, int precision )
   return QString::number( number, 'f', precision );
 }
 
+QString InputUtils::formatDistanceInProjectUnit( const double distanceInMeters, int precision )
+{
+  QgsCoordinateReferenceSystem projectCrs = QgsProject::instance()->crs();
+
```
Comment (PeterPetrik) [inline app/inpututils.cpp:166]: `if !projectCrs.isValid` I would just return `QString::number( distanceInMeters, 'f', precision )` without unit

File: app/inpututils.cpp (hunk 1 above)
Comment (PeterPetrik) [reply app/inpututils.cpp:172]: also `QgsUnitTypes::encodeUnit` for units

File: app/inpututils.cpp [hunk 3]
```diff
@@ -160,6 +160,35 @@ QString InputUtils::formatNumber( const double number, int precision )
   return QString::number( number, 'f', precision );
 }
 
+QString InputUtils::formatDistanceInProjectUnit( const double distanceInMeters, int precision )
+{
+  QgsCoordinateReferenceSystem projectCrs = QgsProject::instance()->crs();
```
Comment (PeterPetrik) [inline app/inpututils.cpp:165]: take CRS from `mapSettings->destinationCrs` similar to `InputUtils::angleBetweenGpsAndFeature`, do not use `instance()` singleton if not really needed.

File: app/inpututils.cpp [hunk 4]
```diff
@@ -160,6 +160,22 @@ QString InputUtils::formatNumber( const double number, int precision )
   return QString::number( number, 'f', precision );
 }
 
+QString InputUtils::formatDistanceInProjectUnit( InputMapSettings *mapSettings, const double distanceInMeters, int precision )
+{
+  QgsCoordinateReferenceSystem projectCrs = mapSettings->destinationCrs();
+
+  if ( !projectCrs.isValid() || projectCrs.mapUnits() == Qgis::DistanceUnit::Unknown )
```
Comment (PeterPetrik) [inline app/inpututils.cpp:167]: add  `!mapSettings || `

**Comment by PeterPetrik** [comment]
For testing @jozef-budac 

map canvas crs in meters
override to feets in qgis project settings

image

**Comment by jozef-budac** [comment]
Everything works well

**Comment by PeterPetrik** [comment]
unrelated test failures

**Comment by PeterPetrik** [comment]
@iiLubos can you an unit tests here https://github.com/MerginMaps/input/blob/897285b166a56d47af6c935d96aaa278ad48641f/app/test/testutilsfunctions.cpp#L170

I suggest 
1/ Create function (not Q_INVOCABLE) `String InputUtils::formatDistanceInDistanceUnit( const double distanceInMeters, int  precision, Qgis::DistanceUnit distUnit )`. 
2/ Only test this new function `formatDistanceInDistanceUnit`, so tests are  not dependent on all QgsProject

**Comment by tomasMizera** [comment]
@iiLubos We will merge this one, please add the test in another PR

**Pull request merged**